### PR TITLE
Update actions and fix running twice on a PR

### PR DESCRIPTION
**Pull request by osfanbuff63**
Updates actions and stops the actions from running twice on a PR (in which case now, it will skip the job), and a couple of cosmetic changes that don't do much

### CODE REVIEW

File: .github/workflows/build-and-deploy.yml [hunk 1]
```diff
@@ -30,12 +29,12 @@ jobs:
 
       - name: Deploy
         # Only deploy for commits on 'master'
-        if: github.event_name == 'push' && github.ref == 'refs/heads/master'
-        uses: JamesIves/github-pages-deploy-action@releases/v3
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/master' }}
```
Comment (osfanbuff63) [inline .github/workflows/build-and-deploy.yml:32]: Uses variable syntax now, shouldn't make a difference

File: .github/workflows/build-and-deploy.yml [hunk 2]
```diff
@@ -30,12 +29,12 @@ jobs:
...
+        uses: JamesIves/github-pages-deploy-action@v4
         with:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           BRANCH: gh-pages
           FOLDER: dist
-          GIT_CONFIG_NAME: GitHub Actions
+          GIT_CONFIG_NAME: actions-user
```
Comment (osfanbuff63) [inline .github/workflows/build-and-deploy.yml]: Also doesn't do anything since what is displayed on GitHub, just isn't a bad idea. It inherits from the email first.

File: .github/workflows/build-and-deploy.yml [hunk 3]
```diff
@@ -4,20 +4,19 @@ jobs:
...
 
       - name: Cache Node Modules
-        uses: actions/cache@v1
+        uses: actions/cache@v3
         with:
           path: ~/.npm
           key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
-          restore-keys: |
-            ${{ runner.os }}-node-
-            ${{ runner.os }}-
```
Comment (Marcono1234) [inline .github/workflows/build-and-deploy.yml:20]: Out of curiosity, why did you remove this?

Comment (osfanbuff63) [reply .github/workflows/build-and-deploy.yml:20]: Looking into actions/cache's docs, it isn't needed.

Comment (Marcono1234) [reply .github/workflows/build-and-deploy.yml:20]: I think `${{ runner.os }}-` might indeed not be needed, but `${{ runner.os }}-node-` sounds useful to me in case `package-lock.json` is updated, so it can reuse most of the old cached packages.

The [example](https://github.com/actions/cache/blob/main/examples.md#node---npm) in their repository also includes that. Or to which docs were you referring?

Comment (osfanbuff63) [reply .github/workflows/build-and-deploy.yml:20]: Alright, reverted

**Review by Marcono1234 — APPROVED** [review]
Thanks! Looks reasonable to me.

(Though of course if someone else wants to have a look at this as well, feel free to do so.)

**Review by SPGoding — COMMENTED** [review]
We could also do something similar to mojira/arisa-kt#566 to not run the workflow twice at all instead of skipping one of them.

**Comment by osfanbuff63** [comment]
> We could also do something similar to mojira/arisa-kt#566 to not run the workflow twice at all instead of skipping one of them.

I've done this on repos before, and the advantages of using this is that it will still run on branches that haven't had a PR made yet, or just other branches in general on this repo. It doesn't matter too much here, but it does on some repos.

**Review by SPGoding — APPROVED** [review]
That makes sense!
